lib/pages/share_two.dart

```diff
@@ -0,0 +1,192 @@
+                        ),
+                      ),
+                      radius: 12,
+                      backgroundColor: GREEN,
+                    ),
+                    Expanded(
+                      child: Divider(
+                        color: Colors.black,
+                        height: 80,
+                        thickness: 1,
+                        indent: 20,
+                        endIndent: 20,
+                      ),
+                    ),
+                    CircleAvatar(
+                      child: Text(
+                        getString(context, "step_two"),
+                        style: TextStyle(
+                          color: WHITE,
+                          fontSize: 10,
+                        ),
+                      ),
+                      radius: 12,
+                      backgroundColor: BLUE,
+                    ),
+                  ],
+                ),
+              ),
+              Container(
+                width: MediaQuery.of(context).size.width - 50,
+                child: TextField(
+                  controller: _myController,
+                  decoration: InputDecoration(
+                    hintText: getString(context, "share_email"),
+                    labelText: getString(context, 'search'),
+                    border: OutlineInputBorder(
+                      borderSide: BorderSide(
+                        color: BLUE,
+                      ),
+                    ),
+                    suffixIcon: Icon(
+                      Icons.search,
+                    ),
+                  ),
+                ),
+              ),
+              Padding(
+                padding: EdgeInsets.only(
+                  top: 50.0,
+                  right: 300,
+                ),
+                child: Text(
+                  getString(context, 'my_lists'),
+                ),
+              ),
+              Padding(
+                padding: EdgeInsets.only(
+                  top: 20,
+                ),
+                child: Column(
+                  children: <Widget>[
+                    Container(
+                      height: 100,
+                      child: ListView.builder(
+                        scrollDirection: Axis.horizontal,
+                        itemCount: wishlists.length,
+                        itemBuilder: (BuildContext ctxt, int index) {
+                          return GestureDetector(
+                            onTap: () {
+                              _listSelected = wishlists[index];
+                              if (searchTermEmail != null) {
+                                shareProvider.addSharedToDataBase(
+                                    searchTermEmail, _listSelected);
+                                openSharePage();
+                              } else {
+                                shareProvider.addSharedToDataBase(
+                                    contactSelected.emails.elementAt(0).value,
```

Comment:
Bug here. If the selected contact does not have any email this line throws an Exception. You should be sure to filter contacts of ShareOne with only contacts with an email.